wafer/users/views.py

```diff
@@ -87,7 +96,10 @@ class RegistrationView(EditOneselfMixin, FormView):
     template_name = 'wafer.users/registration.html'
 
     def get_user(self):
-        return UserProfile.objects.get(user__username=self.kwargs['username'])
+        try:
+            return UserProfile.objects.get(user__username=self.kwargs['username'])
+        except UserProfile.DoesNotExist:
+            raise Http404
```

Comment:
The reason isn't shown to anyone unless the `DEBUG=True` (at least according to the Django docs -- https://docs.djangoproject.com/en/1.9/topics/http/views/#the-http404-exception).